### PR TITLE
Super class generics failing to find return type

### PR DESCRIPTION
**Pull request by cmercer**
#311

### CODE REVIEW

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java [hunk 1]
```diff
@@ -87,13 +90,37 @@ protected FieldDescriptors createFieldDescriptors(Operation operation,
...
+                return findTypeFromTypeVariable(typeVariable, param.getContainingClass());
+            }             
             return ((ParameterizedType) type).getActualTypeArguments()[0];
         } else {
             return Object.class;
         }
     }
 
+    protected Type findTypeFromTypeVariable(TypeVariable typeVariable, Class<?> clazz) {
+        Type defaultReturnValue = Object.class;
```
Comment (fbenz) [inline spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java]: This is only used at the bottom in the last return statement. I would recommend avoiding the variable completely and maybe add a comment. Alternatively, the variable could be moved to the return statement.

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java [hunk 2]
```diff
@@ -87,13 +90,37 @@ protected FieldDescriptors createFieldDescriptors(Operation operation,
 
     protected Type firstGenericType(MethodParameter param) {
         Type type = param.getGenericParameterType();
-        if (type instanceof ParameterizedType) {
+        if(type instanceof TypeVariable) {
```
Comment (fbenz) [inline spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java]: Your formatting is a bit inconsistent with regards to space after `if` or `for`. In this project, we always place a space behind it.

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java (hunk 1 above)
Comment (cmercer) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java]: Updated with better formatting

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java (hunk 2 above)
Comment (cmercer) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java]: Updated formatting

**Pull request merged**